### ISSUE TITLE
azurerm_security_center_workspace crashes when creating or updating workspace

### ISSUE THREAD

**Issue by janasikdawid** [terraform-providers/terraform-provider-azurerm#9700]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_2.39.0`
* `terraform_0.14.0`
* `template_2.2.0`
* `azuread_1.1.1`
* `null_3.0.0`

### Terraform Configuration Files

```hcl
`resource "azurerm_security_center_workspace" "asc_workspace_hub" {
  scope        = "/subscriptions/${var.hub_subscription_id}"
  workspace_id = var.log_analytics_workspace.id
}`
```

### Panic Output
No Panic Output

### Expected Behaviour

Terraform should create / update Azure Security Center Workspace as per plan:
`module.azure_security_center.azurerm_security_center_workspace.asc_workspace_hub will be created
resource "azurerm_security_center_workspace" "asc_workspace_hub" {
id           = (known after apply)
scope        = "/subscriptions/<subscription-id>"
workspace_id = "/subscriptions/<subscription-id>/resourceGroups/rg-monitoring-hub-001/providers/Microsoft.OperationalInsights/workspaces/log-monitor-hub-001"
}`

### Actual Behaviour

Terraform crashes with bellow error:
`Error: Creating Security Center Workspace: security.WorkspaceSettingsClient#Create: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="LinkedAuthorizationFailed" Message="The client has permission to perform action 'Microsoft.Resources/subscriptions/rg-monitoring-hub-001/Microsoft.OperationalInsights/log-monitor-hub-001/write' on scope '/subscriptions/<subscription-id>', however the linked subscription 'resourceGroups' was not found. "`
### Steps to Reproduce

1. `terraform apply`

**Comment by grillba** [comment]
Looks related to the changes in #9651 - pinning to 2.38.0 resolves it

**Comment by janasikdawid** [comment]
> Pinning to 2.38.0 resolves it

I use 2.39.0 which should contain mentioned fix but it is still there.

**Comment by janasikdawid** [comment]
Problem has been resolved in 2.41.0 version
